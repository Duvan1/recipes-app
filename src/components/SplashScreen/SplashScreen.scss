.splash-screen {
    position: fixed;
    top: 0;
    left: 0;
    width: 100vw;
    height: 100vh;
    background-color: rgba(255, 255, 255, 0.9);
    display: flex;
    justify-content: center;
    align-items: center;
    z-index: 1000;
  
    &__spinner {
      width: 50px;
      height: 50px;
      border: 5px solid rgba(0, 0, 0, 0.1);
      border-top: 5px solid #d84a4a;
      border-radius: 50%;
      animation: spin 1s linear infinite;
    }
  
    @keyframes spin {
      0% {
        transform: rotate(0deg);
      }
      100% {
        transform: rotate(360deg);
      }
    }
  }
  