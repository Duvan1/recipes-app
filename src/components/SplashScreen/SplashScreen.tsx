import React from 'react';
import './SplashScreen.scss';

const SplashScreen: React.FC = () => {
  return (
    <div className="splash-screen" role="status">
      <div className="splash-screen__spinner"></div>
    </div>
  );
};

export default SplashScreen;
