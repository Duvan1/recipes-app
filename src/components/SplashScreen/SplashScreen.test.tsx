import { render, screen } from '@testing-library/react';
import SplashScreen from './SplashScreen';

describe('SplashScreen Component', () => {
  it('renders the splash screen container', () => {
    render(<SplashScreen />);

    // Verifica que el contenedor principal se renderiza
    const splashScreen = screen.getByRole('status');
    expect(splashScreen).toHaveClass('splash-screen');
  });

  it('renders the spinner inside the splash screen', () => {
    render(<SplashScreen />);

    // Verifica que el spinner se renderiza dentro del contenedor
    const spinner = screen.getByRole('status').firstChild;
    expect(spinner).toHaveClass('splash-screen__spinner');
  });

  it('matches the snapshot', () => {
    const { container } = render(<SplashScreen />);
    expect(container).toMatchSnapshot();
  });
});
