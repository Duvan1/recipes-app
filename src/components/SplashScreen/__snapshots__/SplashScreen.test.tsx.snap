// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashScreen Component matches the snapshot 1`] = `
<div>
  <div
    class="splash-screen"
  >
    <div
      class="splash-screen__spinner"
    />
  </div>
</div>
`;
